tests/functional/modules/test_kubernetesmod.py

```diff
@@ -0,0 +1,881 @@
+    """
+    res = kubernetes.nodes()
+    assert isinstance(res, list)
+    assert len(res) > 0
+    assert any("salt-test-control-plane" in node for node in res)
+
+
+def test_kubernetes_namespaces(kubernetes):
+    """
+    Test kubernetes.namespaces returns list of namespaces
+    """
+    res = kubernetes.namespaces()
+    assert isinstance(res, list)
+    assert "default" in res, "Default namespace not found"
+    assert "kube-system" in res, "kube-system namespace not found"
+
+
+def test_kubernetes_pods(kubernetes):
+    """
+    Test kubernetes.pods returns list of pods in kube-system namespace
+    """
+    res = kubernetes.pods(namespace="kube-system")
+    assert isinstance(res, list)
+    assert len(res) > 0
+
+
+def test_create_namespace_twice(kubernetes, _cleanup):
+    """
+    Test creating a namespace that already exists raises the appropriate error
+    """
+    test_ns = "salt-test-duplicate-ns"
+
+    # Create namespace first time
+    res = kubernetes.create_namespace(test_ns)
+    assert isinstance(res, dict)
+    assert res["metadata"]["name"] == test_ns
+
+    # Attempt to create same namespace again
+    with pytest.raises(CommandExecutionError, match=".*already exists.*"):
+        kubernetes.create_namespace(test_ns)
+
+    # Cleanup
+    _cleanup("namespace", test_ns)
+
+
+def test_create_namespace_with_invalid_name(kubernetes, caplog):
+    """
+    Test creating a namespace with an invalid name raises appropriate error
+    Names must be lowercase RFC 1123 labels (no underscores or uppercase)
+    """
+    caplog.set_level(logging.INFO)
+    invalid_name = "under_score"
+
+    with pytest.raises(CommandExecutionError) as exc:
+        kubernetes.create_namespace(invalid_name)
+    assert "Invalid" in str(exc.value)
+
+
+def test_delete_system_namespace(kubernetes):
+    """
+    Test attempting to delete protected system namespaces raises error
+    """
+    protected_namespaces = ["default", "kube-system", "kube-public"]
+
+    for namespace in protected_namespaces:
+        with pytest.raises(CommandExecutionError) as exc:
+            kubernetes.delete_namespace(namespace)
+        assert "forbidden" in str(exc.value).lower()
+
+
+def test_list_namespaces_filtering(kubernetes, _cleanup):
+    """
+    Test listing namespaces shows newly created ones
+    and doesn't show deleted ones after deletion
+    """
+    test_ns = "salt-test-filtering"
+
+    # Create namespace and verify it appears in list
+    kubernetes.create_namespace(test_ns)
+
+    # Cleanup namespace
+    _cleanup("namespace", test_ns)
+
+
+@pytest.mark.parametrize(
+    "case",
+    [
+        {
+            "name": "salt-test-plaintext",
+            "data": {"key": "value"},
+            "expected": "value",
+        },
+        {
+            "name": "salt-test-preencoded",
+            "data": {"key": "dmFsdWU="},
+            "expected": "value",
+        },
+    ],
+)
+def test_create_secret_inputs(case, kubernetes, _cleanup):
```

Comment:
```suggestion
@pytest.mark.parametrize(
    "name,data,expected",
    [
        (
            "salt-test-plaintext",
            {"key": "value"},
            "value",
        ),
        (
            "salt-test-preencoded",
            {"key": "dmFsdWU="},
            "value",
        ),
    ],
)
def test_create_secret_inputs(name, data, expected, kubernetes, _cleanup):
```

suggestion: Parametrize the values directly (applies to all parametrizations with `case`).